Added docs for the bulk deployment creation feature (#2000)
OctopusDeploy/docs: src/pages/docs/deployments/bulk-deployment-creation.md

---
layout: src/layouts/Default.astro
pubDate: 2023-09-13
modDate: 2023-09-13
title: Bulk deployment creation
description: Octopus Deploy can use bulk deployment creation when deploying to multiple environments or tenants
navOrder: 160
---

:::div{.hint}
Bulk deployment creation will be available from Octopus Deploy **2023.4**
:::

With bulk deployment creation, you will be redirected to a new server task upon deploying a release to multiple environments or tenants through Octopus Portal. There you will find each requested deployment as task summary items, which will display progress and any errors.

> 

Author: ```suggestion
```

> In the bulk deployment creation server task, you can see the progress of deployments being created and any errors during creation.

Author: ```suggestion
```

---
layout: src/layouts/Default.astro
pubDate: 2023-09-13
modDate: 2023-09-13
title: Bulk deployment creation
description: Octopus Deploy can use bulk deployment creation when deploying to multiple environments or tenants
navOrder: 160
---

:::div{.hint}
Bulk deployment creation will be available from Octopus Deploy **2023.4**
:::

> When deploying a release to multiple environments or tenants through the Octopus Portal, the bulk deployment creation feature will immediately take you to a new server task which will create a deployment for each environment or tenant you are deploying to.

Reviewer: Consider re-wording to avoid passive voice:

You will be redirected to the server task page upon deploying a release to multiple environments or tenants. There you will find each requested deployment   recorded as an indvidual task.

Author: A few thoughts: 

* Worth calling out that this is an Octopus Portal feature. 
* I was reluctant to introduce the term "requested" as a concept but given that we use it in code, it does make comms easier all round so I'm happy to use it.
* Worth disambiguating "deployment recorded as an indvidual task". I think server task when I see the word "task" and the user really just sees a summary

What about:

```suggestion
With bulk deployment creation, you will be redirected to a new server task upon deploying a release to multiple environments or tenants through Octopus Portal. There you will find each requested deployment as task summary items, which will display progress and any errors.
```

> In the bulk deployment creation server task, you can see the progress of deployments being created and any errors during creation.

Reviewer: Can consolidate this now with above:

There you will find each requested deployment recorded as an indvidual task, which will display progress and any errors.

:::figure
![](/docs/deployments/bulk-deployment-creation.png)
:::

You can also navigate away from the server task and it will continue creating deployments in the background.

> If you need to return to a bulk deployment creation server task, you can go to the **Tasks** page and filter down to **Bulk deployment creation** task types.

Reviewer: Consider minor re-wording:

If you need to return to a bulk deployment creation server task, you can `navigate` to the **Tasks** page and use the task filter to select **Bulk deployment creation** task types.

Author: ```suggestion
If you need to return to a bulk deployment creation server task, you can navigate to the **Tasks** page and use the task filter to select **Bulk deployment creation** task types.
```

# Retrying deployment creation failures

> If a deployment to an environment or tenant fails to create, you can **Re-run** the bulk deployment creation server task. This will create a new server task that reattempts creating deployments for the environments or tenants that failed.

Reviewer: Consider rewording to avoid the passive voice to something like:

---
There are two options to retry failed requested deployments. 

**Retrying individual failed deployments**:
Navigate to the task page and click the failed deployment task you wish to retry. Click the re-run button.

**Retrying all failed deployments**:
Navigate to the task page and click the bulk deployment task with the failed deployments. Click the re-run button. This will create a new server task the attempts to deploy only those deployments that failed previously.

---

Or however else you'd like.

Author: > **Retrying individual failed deployments**:
Navigate to the task page and click the failed deployment task you wish to retry. Click the re-run button.

Can't actually do the above but I take your point. 

What about:

```suggestion
To retry failed requested deployments, navigate to the bulk deployment server task with the failed requested deployments and click **Re-run**. This will create a new server task that attempts to recreate only those deployments that failed to create previously.
```

:::figure
![](/docs/deployments/bulk-deployment-creation-retry.png)
:::

# Permissions

> To create, view or re-run a bulk deployment creation server task, your permissions will need to be scoped to at least all the environments and tenants that a bulk deployment creation server task is creating deployments for.

Reviewer: Consider rewording to avoid the passive voice:

Permissions will need to be scoped to all requested environments and tenants when creating, viewing, or re-running a bulk deployment.

Author: I feel "requested" could be ambiguous though as it's not clear what requested is referring to.

Author: ```suggestion
Permissions will need to be scoped to all requested environments and tenants when creating, viewing, or re-running a bulk deployment server task.
```

For example, imagine a bulk deployment creation server task that creates deployments in the 
* Car Rental project 
* To the Development environment 
* For tenants Norfolk and Des Moines

> You'll only be able to perform operations against this server task if your permissions are scoped to *at least* the Car Rental project, the Development environment and Norfolk and Des Moines tenants.

Reviewer: Consider removing contractions in technical writing. public documentation shouldn't be in casual or informal prose.

Reviewer: Consider rewording to consolidate the following paragraph (it generally doesn't make sense to me - what do mean by 'perform operations agagisnt this server task'?):

---

You will only be allowed to create deployments if your user has permissions scoped to the project, environment, and all tenants. However, you only need permissions scoped to the project in order to _view_ the requested deployments on the**Tasks** page.

Author: > Consider removing contractions in technical writing. public documentation shouldn't be in casual or informal prose.

I thought Octopus public docos did this deliberately as there's numerous examples of this but happy to remove contractions.

Author: Operations refers to any CRUD operation against the bulk deployment server task. eg. creating, retrying, editing the state, cancelling etc.

What about 

```suggestion
You will only be allowed to view, retry, cancel or edit the server task if your user has permissions scoped to the project, environment, and all tenants. However, you only need permissions scoped to the project to view the server task on the **Tasks** page.
```

> 

Author: ```suggestion
```

> There is one exception to the above. You only need sufficient project scoping on your permissions to view a bulk deployment server task on the **Tasks** page.

Reviewer: I would remove this paragraph

> There is one exception to the above. You only need sufficient project scoping on your permissions to view a bulk deployment server task on the **Tasks** page.

Author: ```suggestion
```
